### PR TITLE
Move nconnect codes into methods of nconnect struct

### PR DESCRIPTION
**Pull request by billfort**

### CODE REVIEW

File: nconnect.go [hunk 1]
```diff
@@ -72,12 +78,29 @@ func Run(opts *config.Opts) {
...
+		return nil, err
 	}
 
 	account, err := nkn.NewAccount(seed)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
+	}
+
+	if opts.Address {
```
Comment (yilunzhang) [inline nconnect.go:89]: Changing orders will cause problems. Address will not be saved if `opts.Address` or `opts.WalletAddress` is true

File: nconnect.go [hunk 2]
```diff
@@ -91,31 +114,15 @@ func Run(opts *config.Opts) {
 		persistConf.Identifier = config.RandomIdentifier()
 		opts.Identifier = persistConf.Identifier
 		shouldSave = true
+	} else {
+		persistConf.Identifier = opts.Identifier
```
Comment (yilunzhang) [inline nconnect.go:118]: Why modify persistConf when it should not be saved?

File: nconnect.go [hunk 3]
```diff
@@ -245,234 +244,271 @@ func Run(opts *config.Opts) {
 		ssConfig.UDPSocks = true
 	}
 
-	var tun *tunnel.Tunnel
+	nc := &nconnect{opts: opts, account: account, clientConfig: clientConfig, tunnelConfig: tunnelConfig, ssConfig: ssConfig,
```
Comment (yilunzhang) [inline nconnect.go:247]: Line too long, better to break it, one field per line. See ssConfig at L233

File: nconnect.go [hunk 4]
```diff
@@ -245,234 +244,271 @@ func Run(opts *config.Opts) {
...
-					}(dest)
-				}
+				}(dest)
 			}
 		}
 	}
 
-	if opts.Server {
-		err = (&opts.Config).VerifyServer()
+	go func() {
```
Comment (yilunzhang) [inline nconnect.go:403]: These two goroutines are duplicated in both StartClient and StartServer functions

File: nconnect.go (hunk 1 above)
Comment (billfort) [reply nconnect.go:89]: Yes, moved after of persistConf.Save()

File: nconnect.go (hunk 2 above)
Comment (billfort) [reply nconnect.go:118]: Restored.

File: nconnect.go (hunk 3 above)
Comment (billfort) [reply nconnect.go:247]: Yes, modified.

File: nconnect.go (hunk 4 above)
Comment (billfort) [reply nconnect.go:403]: Yes. moved into a method.

File: nconnect.go [hunk 5]
```diff
@@ -30,36 +28,45 @@ import (
...
+	persistConf  *config.Config
+
+	adminClientCache *admin.Client
+	remoteInfoCache  *admin.GetInfoJSON
+
+	tunnel   *tunnel.Tunnel
+	tunaNode *types.Node // It is used to connect specific tuna node, mainly is for testing.
+}
 
-	if opts.Client == opts.Server {
```
Comment (yilunzhang) [inline nconnect.go:51]: IMO it's better to do this verification inside `NewNconnect` because other people might call this function too

Comment (billfort) [reply nconnect.go]: Yes, restored.

**Review by yilunzhang — APPROVED** [review]

**Pull request merged**